src/connectors/snowflake/trulens/connectors/snowflake/connector.py

```diff
@@ -119,6 +119,12 @@ def _validate_snowpark_session_with_connection_parameters(
         missing_snowpark_session_parameters = []
         mismatched_parameters = []
         for k, v in snowpark_session_connection_parameters.items():
+            if k in ["account", "user"] and not v:
+                # Streamlit apps may hide these values so we don't check them.
+                # They are required for a Snowpark Session anyway so this isn't
```

Comment:
wouldn't this be "They aren't required"?

Reply:
no they are required.

Basically, AFAICT, you can't create a snowpark session without a user and account. But, in Streamlit apps, when we ask for them, they hide it from us unless you have a specific setting enabled in your Snowflake account.